src/main/kotlin/com/bugsnag/android/gradle/internal/ExternalNativeBuildTaskUtil.kt

```diff
@@ -0,0 +1,29 @@
+package com.bugsnag.android.gradle.internal
+
+import com.android.build.gradle.tasks.ExternalNativeBuildTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+import java.io.File
+import kotlin.reflect.full.memberProperties
+
+class ExternalNativeBuildTaskUtil(private val providerFactory: ProviderFactory) {
+    private fun getSearchDir(buildTask: Provider<ExternalNativeBuildTask>, propName: String): Provider<File> =
+        buildTask.flatMap { task ->
+            val soFolder = ExternalNativeBuildTask::class.memberProperties.find { it.name == propName }?.get(task)!!
```

Comment:
Does the unwrapped result here (`!!`) represent failure cases that can happen / should be handled?

Reply:
Not currently, both variants of the property are non-nullable in Kotlin.